### DISCUSSION TITLE
outlook.com breaks magic links

### DISCUSSION THREAD

**Discussion by tttp** [supabase/supabase#15455] [Questions]
Hi,

we have some users that can't login by email (magic link). it seems that the common pattern is that they are all using outlook.com as their email provider.
the error they get is:

    UNAUTHORIZED_CLIENT: Email link is invalid or has expired, 

They assure me they get that on the first time they click on the link. Would it be possible that outlook does automatically visit each link on the email and is seen as a login click by supabase/gotrue? 

I'd like to investigate more before creating an issue, could you help me finding out if our hypothesis is correct or if it's due to something else?

**Comment by GaryAustin1** [comment]
Yes, this is a known issue with magic links...
https://supabase.com/docs/guides/auth/auth-email-templates#limitations

_Certain email providers may have spam detection or other security features that prefetch URL links from incoming emails.
In this scenario, the {{ .ConfirmationURL }} sent will be consumed instantly which leads to a "Token has expired or is invalid" error.
To guard against this:
    Use an email OTP instead by including {{ .Token }} in the email template.
    Create your own custom email link to redirect the user to a page where they can click on a button to confirm the action.
    ... more at link_

**Reply by tttp** [reply]
Thanks, it clarifies 

would it be possible to consider changing the default behaviour of gotrue to either
-  ignore the clicks from bot agents or
- have the confirmationURL being an empty page with a js action that authenticate (assuming the outlook bot won't interpret javascript)

**Reply by GaryAustin1** [reply]
As to your new questions you might generate those in Supabase/gotrue issues.   I’m not a Supabase dev and can’t comment on what they might be able to change.

**Comment by drummerjolev** [comment]
[Duplicate](https://github.com/orgs/supabase/discussions/3961)

Having the same issue..
